### PR TITLE
Backfill feature tag to all telemetry tests

### PR DESCRIPTION
**Pull request by bianpengyuan**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pkg/test/framework/features/features.yaml [hunk 1]
```diff
@@ -18,7 +18,18 @@ features:
...
+      stats:
+        prometheus:
+          merge:
+          http:
+            nullvm:
+            wasm:
+          tcp:
+      attributegen:
+      tracing:
+        client-tracing:
```
Comment (howardjohn) [inline pkg/test/framework/features/features.yaml:30]: should we have tracing.client-tracing or tracing.client? Not sure which makes more sense @therealmitchconnors

Comment (bianpengyuan) [reply pkg/test/framework/features/features.yaml]: Ah, yes, there is clear a dup in naming. I remove `-tracing` from the leaf names.

File: pkg/test/framework/features/features.yaml [hunk 2]
```diff
@@ -18,7 +18,18 @@ features:
   # features that relate to measuring or exploring your service mesh or the services which comprise it.
   observability:
     telemetry:
-    tracing:
+      stats:
```
Comment (douglas-reid) [inline pkg/test/framework/features/features.yaml:21]: curious: how do we explore dimensions of features (CNI, VM, Multicluster)?  

and where do things like "customizable metrics" and "remote extensions" stuff plugin on this list?

Comment (bianpengyuan) [reply pkg/test/framework/features/features.yaml:21]: How about `observability.telemetry.stats.env.{CNI, VM, Multicluster}`?

I added customizable metrics into feature list and make request classification test cover both request-classfication and metric customization.

For extension related stuff, there is actually a section `extensibility` at the bottom of the list, remote load would be a good fit there.

Comment (mandarjog) [reply pkg/test/framework/features/features.yaml:21]: That is a good question.  `env` is a cross cutting concern. 
I expect to ask a question like "Are all telemetry features being covered in multi cluster testing" 
There is multi cluster / multi network, multi cluster flatnetwork?Should we try to encode this info or provide some other way.
@douglas-reid @therealmitchconnors

Comment (therealmitchconnors) [reply pkg/test/framework/features/features.yaml:21]: Sorry I missed this thread.  Environment is measured separately from the feature label.  Currently, we track environments like MultiCluster and VM (this is still in progress).  Can you point me to an example of a CNI test, and I will investigate how to add that to the environment dimension?

**Pull request merged**